js_modules/dagit/packages/core/src/runs/Run.tsx

```diff
@@ -198,9 +206,17 @@ const RunWithData: React.FC<RunWithDataProps> = ({
       });
       setComputeLogKey(selectedLogKey || logFileKeys[0]);
     } else if (!stepKeys.includes(computeLogKey)) {
-      setComputeLogKey(selectionStepKeys.length === 1 ? selectionStepKeys[0] : stepKeys[0]);
+      const matching = matchingComputeLogKeyFromStepKey(
+        Object.values(metadata.logCaptureSteps || []),
+        selectionStepKeys.length === 1 ? selectionStepKeys[0] : stepKeys[0],
+      );
+      matching && setComputeLogKey(matching);
```

Comment:
It looks like the computeLogKey is already `logFileKey` in the URL query string -- it might be nice to go ahead and rename it `setComputeLogFileKy` here. I sort of expected it to be a step key until I got to the `?.fileKey` bit